src/node/build/index.ts

```diff
@@ -771,12 +771,14 @@ function createEmitPlugin(
  */
 function getEmitIndexPath(
   indexPath: string,
+  root: string,
   outDir: string,
   { input, output }: Build
 ) {
-  return typeof input === 'string' && path.resolve(input) === indexPath
-    ? path.resolve(outDir, output.file || input)
-    : ''
+  if (typeof input === 'string' && path.resolve(root, input) === indexPath) {
```

Comment:
The `input` option of Rollup is relative to the working directory. Resolving relative to Vite's root would be incorrect.